detection_rules/rule_loader.py

```diff
@@ -153,6 +155,155 @@ def filter(self, cb: Callable[[DeprecatedRule], bool]) -> 'RuleCollection':
         return filtered_collection
 
 
+class RawRuleCollection(BaseCollection):
+    """Collection of rules in raw dict form."""
+
+    __default = None
+    __default_bbr = None
+
+    def __init__(self, rules: Optional[List[dict]] = None, ext_patterns: Optional[List[str]] = None):
+        """Create a new raw rule collection, with optional file ext pattern override."""
+        # ndjson is unsupported since it breaks the contract of 1 rule per file, so rules should be manually broken out
+        # first
+        self.ext_patterns = ext_patterns or ['*.toml', '*.json']
+        self.id_map: Dict[definitions.UUIDString, DictRule] = {}
+        self.file_map: Dict[Path, DictRule] = {}
+        self.name_map: Dict[definitions.RuleName, DictRule] = {}
+        self.rules: List[DictRule] = []
+        self.errors: Dict[Path, Exception] = {}
+        self.frozen = False
+
+        self._raw_load_cache: Dict[Path, dict] = {}
+        for rule in (rules or []):
+            self.add_rule(rule)
+
+    def __contains__(self, rule: DictRule):
+        """Check if a rule is in the map by comparing IDs."""
+        return rule.id in self.id_map
+
+    def filter(self, cb: Callable[[DictRule], bool]) -> 'RawRuleCollection':
+        """Retrieve a filtered collection of rules."""
+        filtered_collection = RawRuleCollection()
+
+        for rule in filter(cb, self.rules):
+            filtered_collection.add_rule(rule)
+
+        return filtered_collection
+
+    def _load_rule_file(self, path: Path) -> dict:
+        if path in self._raw_load_cache:
+            return self._raw_load_cache[path]
+
+        if path.suffix == ".toml":
+            # use pytoml instead of toml because of annoying bugs
+            # https://github.com/uiri/toml/issues/152
+            # might also be worth looking at https://github.com/sdispater/tomlkit
+            raw_dict = pytoml.loads(path.read_text())
+        elif path.suffix == ".json":
+            raw_dict = json.loads(path.read_text())
+        elif path.suffix == ".ndjson":
+            raise ValueError('ndjson is not supported in RawRuleCollection. Break out the rules individually.')
+        else:
+            raise ValueError(f"Unsupported file type {path.suffix} for rule {path}")
+
+        self._raw_load_cache[path] = raw_dict
+        return raw_dict
```

Comment:
Can we result the utiles methods for some of these methods under `RawRuleCollection` like utils.load_rule_contents`? It seems almost as if there's potential overlap with that `BaseCollection` should be.

Reply:
out of scope for this PR